### PR TITLE
Add jobs to semi-automate publication of @webref/idl package

### PR DESCRIPTION
**Pull request by tidoust**
This roughly implements the logic discussed in https://github.com/w3c/webref/issues/123#issuecomment-796650510:

1. The `prepare-idl-release` job runs on Thursday morning. It compares the latest contents of the repo with the published `@webref/idl` package on NPM, and creates a PR that bumps the version of the package in `packages/idl/package.json`, and exposes the diff of the IDL.

2. The `release-idl` job runs whenever an update to `packages/idl/package.json` gets pushed to the default branch. It publishes a new @webref/idl package to npm and creates a new @webref/idl release on GitHub. The job is a no op if the version number in package.json does not differ from the one published on npm.

If all goes well, merging the PR created by the first job will trigger the second job.

The update also moves the cleanup job to Wednesday morning instead of Monday.

If a patch no longer applies when the first job runs, it will fail. It won't create an issue in such cases, but repo admins will receive a "job failed" notification.

To be entirely clear, I have no idea whether these jobs are going to work as expected :) That said, at worst, they should simply do nothing and we should be able to iterate progressively to get them to do something useful.

If things work, the logic can easily be duplicated for the `@webref/css` package.

### CODE REVIEW

File: .github/workflows/prepare-idl-release.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    - name: Compute diff between new release and current package
+      # For details on syntax to use to export a multiline string to an
+      # environment variable, see:
+      # https://github.com/actions/toolkit/blob/main/docs/commands.md#set-an-environment-variable
+      # The diff does not take the package.json file into account because
+      # "npm install" adds properties that start with "_" to that file which do
+      # not exist in the repo version of the file.
+      run: |
+        echo 'IDL_DIFF<<EOF' >> $GITHUB_ENV
+        diff idl/node_modules/@webref/idl webref/packages/idl --ignore-trailing-space --exclude=package.json >> $GITHUB_ENV
```
Comment (foolip) [inline .github/workflows/prepare-idl-release.yml:58]: Is the path here right? With no `working-directory` here, what's the implicit working directory? I would have guessed it's the default location for a checkout, not the parent and this seems to assume.

File: .github/workflows/release-idl.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    - name: Publish npm package
+      id: publish
+      uses: JS-DevTools/npm-publish@v1
+      with:
+        token: ${{ secrets.NPM_TOKEN }}
+        package: ./packages/idl/package.json
+        access: public
+
+    - name: Create GitHub release
```
Comment (foolip) [inline .github/workflows/release-idl.yml:46]: I didn't create a GitHub release for any of the earliest releases, what happens if there is none, or if this last step sometimes fails?

**Review by foolip — APPROVED** [review]
LGTM with questions.

**Review by dontcallmedom — APPROVED** [review]

File: .github/workflows/prepare-idl-release.yml (hunk 1 above)
Comment (tidoust) [reply .github/workflows/prepare-idl-release.yml:58]: I do not really know but isn't that the same thing as in the `update-ed` job?
https://github.com/w3c/webref/blob/master/.github/workflows/update-ed.yml#L37

Comment (foolip) [reply .github/workflows/prepare-idl-release.yml:58]: Perhaps, I didn't compare it to anything. Let's try and see if it works?

File: .github/workflows/release-idl.yml (hunk 2 above)
Comment (tidoust) [reply .github/workflows/release-idl.yml:46]: I'm confused by the Releases page on GitHub:
https://github.com/w3c/webref/releases

I'm the one who created a release for the CSS package. The page also lists IDL package releases, but perhaps they are just tags created with `git tag`? If you did not create the tags yourself, then Lerna most likely did.

I'll push an update to the PR to simply create a tag for now. It seems good practice to tag the commit associated with a particular npm package. We don't need something more fancy for now.

Comment (tidoust) [reply .github/workflows/release-idl.yml]: Done in last commit. If these steps fail, then that means the npm package will not be directly associated with a commit. We should be notified of the job's failure and can always add the tag manually afterwards.

**Comment by tidoust** [comment]
FYI, the preparation job fails at the diff step. It may be due to the "echo" syntax (which I don't necessarily understand but copied over from the [doc](https://github.com/actions/toolkit/blob/main/docs/commands.md#set-an-environment-variable)). Or perhaps simply because the diff command fails (e.g. because the paths are incorrect). I'll make a couple of updates directly on the default branch to isolate the error, then watch, rinse and repeat...


The reported error is:
```
> Run echo 'IDL_DIFF<<EOF' >> $GITHUB_ENV
  echo 'IDL_DIFF<<EOF' >> $GITHUB_ENV
  diff idl/node_modules/@webref/idl webref/packages/idl --ignore-trailing-space --exclude=package.json >> $GITHUB_ENV
  echo 'EOF' >> $GITHUB_ENV
  shell: /usr/bin/bash -e {0}
Error: Process completed with exit code 1.
Error: Unable to process file command 'env' successfully.
Error: Invalid environment variable value. Matching delimiter not found 'EOF'
```

**Comment by tidoust** [comment]
In practice, everything works as expected. Problem is that, when there is a diff, the `diff` command outputs the diff but also exits with a `1` exit code, which gets intepreted as failure. What is the easiest way to ignore the exit code?

**Comment by dontcallmedom** [comment]
I don't know if it's the most recommendable way, but `diff foo bar || echo -n` is guaranteed not fail

**Comment by tidoust** [comment]
That seems to work, Job created https://github.com/w3c/webref/pull/138, which seems correct.

Remaining verification steps:
- merge that PR to check the release job.
- make sure that the prepare job does not create a PR when there are no diffs.

**Comment by tidoust** [comment]
Yeah, NPM publication works! See https://www.npmjs.com/package/@webref/idl/v/1.0.9
Creation of a tag failed because the committer name needs to be specified, see https://github.com/w3c/webref/pull/139

**Pull request merged**